### ISSUE TITLE
Add extra options for custom format.

### ISSUE THREAD

**Issue by FakeKitten** [igoticecream/Snorlax#17]
Adding the option for move types has already been suggested here but in addition to that it would be nice to see;
- HP: this would be most useful as 3 different possible strings. Total HP %HPT%, Current HP %HPC%, and HP remaining %HPR% The last of these would be the most useful honestly as it would serve as a quick and accurate way to know which potions to use for what pokes without wastage.
- Candy %CDY%: simply shows the amount of candy currently available for  specific pokemon. ie bulbasaur, ivysaur and venusaur would all show the current amount of bulbasaur candy the player has. Useful for mass evolutions.
- Nick %NICK%:- In it's current state this shows the pokemon's original name which isn't optimal and I would like to see this show what a pokemon has been renamed as. %NAME% may be a better string to use for showing a pokemons name if that feature is kept.
- Moveset perfection %MOP%: this is a rather complex one. This would show how good a moveset is for a pokemon compared to the best possible moveset as a percentage. ie a hydro pump vaporeon would show 100% but with water pulse would be 78%
  For this I'm taking combined fast+charge STAB DPS from [this site](https://pokemon-go-dps-optimizer.herokuapp.com/?search=vaporeon) and taking the highest one as 100% for any specific pokemon then calculating the rest by dividing their stab DPS by the optimal stab DPS and multiplying by 100.  But I'm not entirely sure how accurate those figures are and if there's a better way to calculate the perfection of a moveset.

**Comment by likebutta222** [comment]
I agree with most things here but I think the most important one would be to retain the user's nickname.  Perhaps, right now %NICK% can be reserved for original name as is and %CUSTOM% can be used to retain the user's custom nickname.  I do this primarily for differentiation reasons when choosing which pokemon to battle/defend with, even if they are comparable IV/CP.

Thank you very much.  I appreciate your efforts and a donation is on the way shortly.

**Comment by justmiike** [comment]
Yep, I agree, this was the feature I am looking for:

%NICK% can be the original name and %CUSTOM% can be the nickname users set.

So:
%NICK% %IV%% - Pikachu 89%
%CUSTOM% %IV%% - *YellowMouse 89%

I also add \* at the front of names to indicate best moves and [-] for best moves to defend gyms.

**Comment by igoticecream** [comment]
I like those formats... Hp, Candy and Custom.

And to be honest with you... I won't implement the move perfection in a short/medium term, there are other features i want to implement first.

I'm going to put this on the 1.5.0 milestone

**Comment by cesargomez89** [comment]
What about %SIZE% (XL/XS)?
I would like to know the pokemon's size when I'm attacking or assigning a pokemon to a gym.

**Comment by alexcheveau** [comment]
Lol, why u want to know the Size ?
